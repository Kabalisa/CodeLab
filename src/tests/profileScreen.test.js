import ProfileScreen from '../screens/profileScreen';
describe('ProfileScreen tests', () => {
  const props = {
    navigation: {
      getParam: jest.fn(item => {
        item = {
          node: {
            login: 'name',
            repositories: { totalCount: 5 },
            starredRepositories: { totalCount: 5 }
          }
        };
        return item;
      })
    }
  };
  it('should render without error', () => {
    const wrapper = shallow(<ProfileScreen {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should press on share', () => {
    const wrapper = shallow(<ProfileScreen {...props} />);
    const shareButton = wrapper.find('[id="share"]');
    shareButton.props().onPress();
  });
  it('should press on open-in-app-browser', () => {
    const wrapper = shallow(<ProfileScreen {...props} />);
    const browserButton = wrapper.find('[id="browser"]');
    browserButton.props().onPress();
  });
});
