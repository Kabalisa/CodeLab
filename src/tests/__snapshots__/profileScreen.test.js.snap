// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileScreen tests should render without error 1`] = `
<View>
  <Header
    navigation={
      Object {
        "getParam": [MockFunction] {
          "calls": Array [
            Array [
              "item",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": Object {
                "node": Object {
                  "login": "name",
                  "repositories": Object {
                    "totalCount": 5,
                  },
                  "starredRepositories": Object {
                    "totalCount": 5,
                  },
                },
              },
            },
          ],
        },
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "height": 695,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": undefined,
        }
      }
      style={
        Object {
          "borderRadius": 110,
          "height": 220,
          "marginVertical": 15,
          "width": 220,
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 24,
          "fontWeight": "bold",
          "marginBottom": 25,
        }
      }
    >
      name
    </Text>
    <Text
      style={
        Object {
          "fontSize": 18,
        }
      }
    >
      Github URL:
    </Text>
    <TouchableOpacity
      activeOpacity={0.2}
      id="browser"
      onPress={[Function]}
    >
      <Text
        style={
          Object {
            "color": "#5075D4",
            "fontSize": 18,
            "marginBottom": 20,
          }
        }
      >
        https://github.com/name
      </Text>
    </TouchableOpacity>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginBottom": 20,
        }
      }
    >
      <Text
        style={
          Object {
            "bottom": 6,
            "fontSize": 24,
            "fontWeight": "bold",
            "marginRight": 10,
          }
        }
      >
        5
      </Text>
      <Text
        style={
          Object {
            "fontSize": 18,
          }
        }
      >
        repos
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginLeft": 15,
        }
      }
    >
      <Text
        style={
          Object {
            "bottom": 6,
            "fontSize": 24,
            "fontWeight": "bold",
            "marginRight": 10,
          }
        }
      >
        5
      </Text>
      <Text
        style={
          Object {
            "fontSize": 18,
          }
        }
      >
        starring
      </Text>
    </View>
    <TouchableOpacity
      activeOpacity={0.2}
      id="share"
      onPress={[Function]}
    >
      <View
        style={
          Object {
            "backgroundColor": "#5075D4",
            "borderColor": "#5075D4",
            "borderRadius": 8,
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "center",
            "marginTop": 40,
            "padding": 10,
            "width": 230,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 18,
              "fontWeight": "bold",
            }
          }
        >
          Share Profile
        </Text>
      </View>
    </TouchableOpacity>
  </View>
  <Footer />
</View>
`;
